jumanji/environments/logic/minesweeper/env.py

```diff
@@ -92,46 +95,44 @@ class Minesweeper(Environment[State]):
 
         Args:
-            num_rows: number of rows, i.e. height of the board. Defaults to 10.
-            num_cols: number of columns, i.e. width of the board. Defaults to 10.
-            num_mines: number of mines on the board. Defaults to 10.
+
+            generator: Generator to generate problem instances on environment reset.
+                Implemented options are [`SamplingGenerator`].
+                Defaults to `SamplingGenerator`.
+                The generator will have attributes:
+                    - num_rows: number of rows, i.e. height of the board. Defaults to 10.
+                    - num_cols: number of columns, i.e. width of the board. Defaults to 10.
+                    - num_mines: number of mines generated. Defaults to 10.
             reward_function: `RewardFn` whose `__call__` method computes the reward of an
                 environment transition based on the given current state and selected action.
                 Implemented options are [`DefaultRewardFn`]. Defaults to `DefaultRewardFn`.
             done_function: `DoneFn` whose `__call__` method computes the done signal given the
                 current state, action taken, and next state.
                 Implemented options are [`DefaultDoneFn`]. Defaults to `DefaultDoneFn`.
-            color_mapping: colour map used for rendering.
+            viewer: Viewer to support rendering and animation methods.
+                Implemented options are [`MinesweeperViewer`].
+                Defaults to `MinesweeperViewer`.
         """
-        if num_rows <= 1 or num_cols <= 1:
-            raise ValueError(
-                f"Should make a board of height and width greater than 1, "
-                f"got num_rows={num_rows}, num_cols={num_cols}"
-            )
-        if num_mines < 0 or num_mines >= num_rows * num_cols:
-            raise ValueError(
-                f"Number of mines should be constrained between 0 and the size of the board, "
-                f"got {num_mines}"
-            )
-        self.num_rows = num_rows
-        self.num_cols = num_cols
-        self.num_mines = num_mines
-        self.reward_function = reward_function or DefaultRewardFn()
+        self.reward_function = reward_function or DefaultRewardFn(
+            revealed_empty_square_reward=REVEALED_EMPTY_SQUARE_REWARD,
+            revealed_mine_reward=REVEALED_MINE_REWARD,
+            invalid_action_reward=INVALID_ACTION_REWARD,
+        )
         self.done_function = done_function or DefaultDoneFn()
-
-        self.cmap = color_mapping if color_mapping else COLOUR_MAPPING
-        self.figure_name = f"{num_rows}x{num_cols} Minesweeper"
-        self.figure_size = (6.0, 6.0)
+        self.generator = generator or UniformSamplingGenerator(
+            num_rows=10, num_cols=10, num_mines=10
+        )
+        self._viewer = viewer or MinesweeperViewer(
+            num_rows=self.generator.num_rows, num_cols=self.generator.num_cols
```

Comment:
I would suggest doing this above since you are using these attributes elsewhere in your environment:
```
self.num_rows = self.generator.num_rows
self.num_cols = self.generator.num_cols
```
Then, this becomes:
```
self._viewer = viewer or MinesweeperViewer(self.num_rows, self.num_cols)
```

Reply:
OK